### DISCUSSION TITLE
Web3Modal V3 + ethersV5 -> Programatically switch to a different blockchain network!

### DISCUSSION THREAD

**Discussion by Liberalite** [WalletConnect/walletconnect-monorepo#4016] [Developer Support] [answered]
Hi!

I'm trying to figure out how can we allow users to switch between chains with a custom dropdown instead of the current blockchains modal. I saw that with wagmi it's possible but I could not find a way to do it with ethersV5 or V6. The only thing I think I can do is to inform the user by creating a modal that tells them you are on a different network, please press this button to open the WalletConnect Network Modal and switch to the different chain...

Example that does not work with ethersV5 with the WalletConnect provider:
`

    // Request the user to switch to the target network
        await ethereum.request({
            method: 'wallet_switchEthereumChain',
            params: [{ chainId: targetNetwork.chainId }], // chainId must be in hexadecimal
        });
`

Example with wagmi of programatically changing the network:
`

    import { switchNetwork } from '@wagmi/core'
 
    const network = await switchNetwork({
       chainId: 1,
    })
`
Is there any way to achieve the same thing using ethersV5? 

Would really appreciate if somebody has found a way to do this!

Thank you very much in advance!

**Comment by glitch-txs** [comment, marked as answer]
First example actually works but you need to use the walletProvider returned by Web3Modal function/hook

**Reply by Liberalite** [reply]
Wish that was true for me for Web3Modal V3. 
This is my code:

`

   import { useState, useEffect } from 'react';
   import { ethers } from 'ethers';
   import { useWeb3ModalAccount, useWeb3ModalProvider } from '@web3modal/ethers5/react';

   const useWallet = () => {
      const [ethersProvider, setProvider] = useState<ethers.providers.Web3Provider>();
      const [signer, setSigner] = useState<ethers.providers.JsonRpcSigner>();
      const { address, chainId, isConnected } = useWeb3ModalAccount()
      const { walletProvider } = useWeb3ModalProvider()

   useEffect(() => {
      const run = async() => {
         if(!isConnected || !walletProvider) return
            const ethersProvider = new ethers.providers.Web3Provider(walletProvider)
            setProvider(ethersProvider);
            const signer = await ethersProvider.getSigner()
            setSigner(signer);
         }
         run()
   }, [address, chainId, isConnected])

   return { ethersProvider, signer, address, chainId, isConnected };
   };

   export default useWallet;
   
`

and then in a different react component I want to allow the user to switch to a different chain using this code:

`
  const { address, chainId, isConnected } = useWeb3ModalAccount()
  
 const changeNetwork = async () => {
      try {
        if (!isConnected || !ethersProvider) return;
        // Request the user to switch to the target network
        await ethersProvider.request({
          method: "wallet_switchEthereumChain",
          params: [{ chainId: chainId }], // chainId must be in hexadecimal
        });
      } catch (switchError) {
        // Handle errors, like the user rejecting the request
        console.error("Could not switch to the desired network:", switchError);
      }
};
changeNetwork();
    
  error i get:
  Property 'request' does not exist on type 'Web3Provider'.ts(2339)
⚠ Error (TS2339)  | 
Property request does not exist on type 
 .
any

`

Can you please provide an example?

**Comment by Liberalite** [comment]
Thank you @glitch-txs! 

Switched from Web3Provider to ExternalProvider and now changing blockchains work now 🙏
